server/src/infra/repositories/media.repository.ts

```diff
@@ -70,16 +70,28 @@ export class MediaRepository implements IMediaRepository {
   transcode(input: string, output: string | Writable, options: TranscodeOptions): Promise<void> {
     if (!options.twoPass) {
       return new Promise((resolve, reject) => {
-        ffmpeg(input, { niceness: 10 })
-          .inputOptions(options.inputOptions)
-          .outputOptions(options.outputOptions)
-          .output(output)
-          .on('error', (err, stdout, stderr) => {
-            this.logger.error(stderr);
-            reject(err);
-          })
-          .on('end', resolve)
-          .run();
+        const oldLdLibraryPath = process.env['LD_LIBRARY_PATH'];
+        if (options.ldLibraryPath) {
+          // fluent ffmpeg does not allow to set environment variables, so we do it manually
+          process.env['LD_LIBRARY_PATH'] = (oldLdLibraryPath || '') + options.ldLibraryPath;
```

Comment:
Is there no better way to do this 😭

Also thinking about it, this seems like it has potential for concurrency issues.

Reply:
I know, it's awful. I could raise an issue at fluent-ffmpeg. But not when it even if they add support to set environment variables.

I've read the documentation of the process env in node. I guess there is no concurrency issue because a) the env is reset synchronously without waiting for ffmpeg to run, b) each node worker has its own process env

Reply:
My thought was that we have child processes from exiftool etc. that aren't tied to Node.js, so theoretically when a child process is spawned by it, that process could see either `LD_LIBRARY_PATH` depending on timing. But I'm not sure if this is an issue in practice.

Reply:
well, that child process inherits the env that exiftool had when it was started - and that env is unmodified if the node documentation is correct (see above)

Reply:
Gotcha, it should be fine in that case. 

I'd suggest refactoring this so the ffmpeg code is in its own method and the try finally wraps around a call to it.

Reply:
Actually never mind, it's better if this is handled in the promise so the env gets moved back sooner instead of waiting for the promise to resolve.

Reply:
It would look like this:

```ts
    if (!options.twoPass) {
      return new Promise((resolve, reject) => {
        const oldLdLibraryPath = process.env['LD_LIBRARY_PATH'];
        if (options.ldLibraryPath) {
          // fluent ffmpeg does not allow to set environment variables, so we do it manually
          process.env['LD_LIBRARY_PATH'] = (oldLdLibraryPath || '') + options.ldLibraryPath;
        }
        try {
          this.runFfmpeg(input, output, options, resolve, reject);
        } finally {
          if (options.ldLibraryPath) {
            process.env['LD_LIBRARY_PATH'] = oldLdLibraryPath;
          }
        }
      });
    }
```

and

```ts
  runFfmpeg(
    input: string,
    output: string | Writable,
    options: TranscodeOptions,
    resolve: (value: void | PromiseLike<void>) => void,
    reject: (reason?: any) => void,
  ) {
    ffmpeg(input, { niceness: 10 })
      .setFfmpegPath(options.ffmpegPath || 'ffmpeg')
      .inputOptions(options.inputOptions)
      .outputOptions(options.outputOptions)
      .output(output)
      .on('error', (err, stdout, stderr) => {
        this.logger.error(stderr);
        reject(err);
      })
      .on('end', resolve)
      .run();
  }
```

`runFfmpeg` can't easily be reused for the other two calls, so not sure if this is actually better than the current state. maybe there's something better?

Reply:
found something re-usable

Reply:
Oh nice, I like it